presto-atop/src/main/java/com/facebook/presto/atop/AtopPlugin.java

```diff
@@ -42,25 +38,15 @@ public void setOptionalConfig(Map<String, String> optionalConfig)
     }
 
     @Override
-    public void setTypeManager(TypeManager typeManager)
+    public Iterable<ConnectorFactory> getConnectorFactories(ConnectorFactoryContext context)
     {
-        this.typeManager = typeManager;
-    }
-
-    @Override
-    public void setNodeManager(NodeManager nodeManager)
-    {
-        this.nodeManager = nodeManager;
-    }
-
-    @Override
-    public <T> List<T> getServices(Class<T> type)
-    {
-        NodeConfig nodeConfig = new ConfigurationFactory(optionalConfig).build(NodeConfig.class);
-        if (type == ConnectorFactory.class) {
-            return ImmutableList.of(type.cast(new AtopConnectorFactory(AtopProcessFactory.class, optionalConfig, getClassLoader(), typeManager, nodeManager, nodeConfig)));
-        }
-        return ImmutableList.of();
+        return ImmutableList.of(new AtopConnectorFactory(
+                AtopProcessFactory.class,
+                optionalConfig,
+                getClassLoader(),
+                context.getTypeManager(),
+                context.getNodeManager(),
```

Comment:
Oh I see. You moved them from setters to the context object.  I'm fine with that.